distsql/select_result.go

```diff
@@ -163,6 +217,24 @@ func (r *selectResult) Next(ctx context.Context, chk *chunk.Chunk) error {
 	return nil
 }
 
+func (r *selectResult) readFromArrow(ctx context.Context, chk *chunk.Chunk) error {
+	if r.selectResp == nil || len(r.selectResp.RowBatchData) == 0 {
+		err := r.getSelectResp()
+		if err != nil || r.selectResp == nil {
+			return errors.Trace(err)
+		}
+	}
+	r.readRowBatch(chk)
+	return nil
+}
+
+func (r *selectResult) readRowBatch(chk *chunk.Chunk) {
```

Comment:
I think we don't need to extract a function for this. It's very simple, and the caller is very simple as well.

Reply:
Update.